fabtests/prov/efa/src/efa_rdma_checker.c

```diff
@@ -57,59 +56,66 @@ int main(int argc, char *argv[])
 	device_list = ibv_get_device_list(&dev_cnt);
 	if (dev_cnt <= 0) {
 		fprintf(stderr, "No ibv device found!\n");
-		return -ENODEV;
+		return -FI_ENODEV;
 	}
 
-	ibv_ctx = ibv_open_device(device_list[0]);
-	if (!ibv_ctx) {
-		fprintf(stderr, "cannot open device %d\n", 0);
-		return EXIT_FAILURE;
-	}
+	for (i = 0; i < dev_cnt; i++) {
+		ibv_ctx = ibv_open_device(device_list[i]);
+		if (!ibv_ctx) {
+			fprintf(stderr, "cannot open device %d\n", i);
+			return EXIT_FAILURE;
+		}
 
-	err = ibv_query_device_ex(ibv_ctx, NULL, &ibv_dev_attr);
-	if (!err) {
-		fprintf(stdout, "ibv_dev_attr.device_cap_flags_ex: %lx\n", ibv_dev_attr.device_cap_flags_ex);
-	}
+		memset(&efadv_attr, 0, sizeof(efadv_attr));
+		err = efadv_query_device(ibv_ctx, (struct efadv_device_attr *)&efadv_attr, sizeof(efadv_attr));
+		ibv_close_device(ibv_ctx);
+		if (err) {
+			if (err == EOPNOTSUPP) {
+				fprintf(stdout, "Not an EFA device. Continue to check the next device.\n");
+				continue;
+			} else {
+				fprintf(stderr, "cannot query device %d, err: %d\n", i, -err);
+				goto out;
+			}
+		}
 
-	err = efadv_query_device(ibv_ctx, (struct efadv_device_attr *)&efadv_attr, sizeof(efadv_attr));
-	ibv_close_device(ibv_ctx);
-	if (err) {
-		fprintf(stderr, "cannot query device\n");
-		goto out;
-	}
+		if (efadv_attr.max_rdma_size == 0) {
```

Comment:
This test now checks all of the available devices

I would print a line before this one with the device index (`i`) and name (`device_list[i]->name`) so that it's clear which device the following lines are referring to